In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats # Many functions for probability distributions in python are in scipy.stats
from numpy import random
import os

In [2]:
data = pd.read_csv('A+%26+B+shingles.csv')

In [3]:
data.head()

A     B
0  0.44  0.14
1  0.61  0.15
2  0.47  0.31
3  0.30  0.16
4  0.15  0.37

In [4]:
data.tail()

A   B
31  0.40 NaN
32  0.29 NaN
33  0.43 NaN
34  0.34 NaN
35  0.37 NaN

In [5]:
data.columns

Index(['A', 'B'], dtype='object')

In [6]:
data.isnull().sum()

A    0
B    5
dtype: int64

In [7]:
data.sum()

A    11.40
B     8.48
dtype: float64

In [8]:
data['A'].value_counts().sum()

36

### 3.1 Do you think there is evidence that mean moisture contents in both types of shingles are within the permissible limits? State your conclusions clearly showing all steps.
 

An important quality characteristic used by the manufacturers of ABC asphalt shingles is the amount of moisture the shingles contain when they are packaged. Customers may feel that they have purchased a product lacking in quality if they find moisture and wet shingles inside the packaging.   In some cases, excessive moisture can cause the granules attached to the shingles for texture and coloring purposes to fall off the shingles resulting in appearance problems. To monitor the amount of moisture present, the company conducts moisture tests. A shingle is weighed and then dried. The shingle is then reweighed, and based on the amount of moisture taken out of the product, the pounds of moisture per 100 square feet are calculated. The company would like to show that the mean moisture content is less than 0.35 pounds per 100 square feet.

The file (A & B shingles.csv) includes 36 measurements (in pounds per 100 square feet) for A shingles and 31 for B shingles.

#### Step 1: Define NULL and Alternate Hypothesis.
#### For the A shingles, the null and alternative hypothesis to test whether the population mean moisture content is less than 0.35 pound per 100 square feet is 
#### given:
######  𝐻0≤0.35 
######  𝐻1>0.35

#### For the B shingles, the null and alternative hypothesis to test whether the population mean moisture content is less than 0.35 pound per 100 square feet is

#### given: 
###### 𝐻0≤0.35
###### 𝐻1>0.35

#### Step 2: Write the significance level for shingles A
Here we are given α = 0.05 and it is given that n, sample size = 36.

#### Step 3: Identify the test statistic
t-test one sample shall be used as n=36 and single sample is available.

#### Step 4: Calculate the p - value and test statistic

In [9]:
t_statistic, p_value = stats.ttest_1samp(data['A'], 0.35)
print('One sample t test \nt statistic: {0} p value: {1} '.format(t_statistic, p_value/2))

One sample t test 
t statistic: -1.4735046253382782 p value: 0.07477633144907513 


#### Step 5 Decide to reject or accept null hypothesis

p value is 0.074 and it is greater than 5% level of significance

So statistically we fail to reject the null hypothesis at 5% level of significance.


#### Step 1: Define Null and alternate hypothesis for sample B

#### For the B shingles, the null and alternative hypothesis to test whether the population mean moisture content is less than 0.35 pound per 100 square feet is

#### given: 
###### 𝐻0≤0.35
###### 𝐻1>0.35


#### Step 2: Write the significance level for shingles B
Here we are given α = 0.05 and it is given that n, sample size = 31.

#### Step 3: Identify the test statistic
t-test one sample shall be used as n=31 and single sample is available.

#### Step 4: Calculate the p - value and test statistic

In [10]:
t_statistic, p_value = stats.ttest_1samp(data['B'], 0.35, nan_policy='omit')
print('One sample t test \nt statistic: {0} p value: {1} '.format(t_statistic, p_value/2))

One sample t test 
t statistic: -3.1003313069986995 p value: 0.0020904774003191826 


### Step 5 Decide to reject or accept null hypothesis

p value is 0.002 and it is less than 5% level of significance

So statistically we reject the null hypothesis at 5% level of significance.


### 3.2 Do you think that the population means for shingles A and B are equal? Form the hypothesis and conduct the test of the hypothesis. What assumption do you need to check before the test for equality of means is performed? 

#### Step 1: In testing whether the mean for shingles A and Shingles B are the same, the null hypothesis states that the mean of shingle A to mean of shingle B are the same,  equals . The alternative hypothesis states that the mean are different,  is not equal to 

𝐻0: μA = μB

𝐻1: μA ≠ μB
 

#### Step 2: Write the significance level for shingles B
Here we are given α = 0.05 and the population standard deviation is not known

#### Step 3: Identify the test statistic

We have two samples and we do not know the population standard deviation. Sample sizes for both samples are not the same. The sample size is , n > 30. So we use the t distribution and the 𝑡𝑆𝑇𝐴𝑇 test statistic for two sample test.



#### Step 4: Calculate the p - value and test statistic

In [11]:
t_statistic, p_value  = stats.ttest_ind(data['A'],data['B'], nan_policy='omit')
print('One sample t test \nt statistic: {0} p value: {1} '.format(t_statistic, p_value/2))

One sample t test 
t statistic: 1.2896282719661123 p value: 0.1008748285917653 


In [12]:
t_statistic, p_value  =stats.ttest_ind(data['A'],data['B'],nan_policy='omit')
print('tstat',t_statistic)    
print('P Value',p_value)

tstat 1.2896282719661123
P Value 0.2017496571835306


#### Step 5 Decide to reject or accept null hypothesis

p value is 0.201 and it is greater than 5% level of significance

So statistically we fail to reject the null hypothesis at 5% level of significance.

Two-sample t-test p-value= 0.2017496571835306

Population mean for shingles A and singles B are same


### Quality of Business Report


#### The mean moisture content is less than 0.35 pounds per 100 square feet is applicable only for sample A and not for sample B
#### Population mean for shingles A and singles B are same as per two paired test.
